datastore/sqlite_datastore.nim

```diff
@@ -240,33 +245,35 @@ method get*(
     bytes = dataCol(s, 0)
 
   let
-    exists = ? self.getStmt.query((key.id), onData)
+    queryRes = self.getStmt.query((key.id), onData)
+
+  if queryRes.isErr: return failure queryRes.error.msg
```

Comment:
why not just `return queryRes` - here and in the lot of similar places?

Reply:
I may be able to do that in some places, but in others it wasn't working. I'll check again.

Reply:
Ah, of course, it's because of the type mismatch. The return type (wrapped in `Future`) for `method get` is `?!(?seq[byte])` while for `proc query` it's `?!bool.

If we do `if queryRes.isErr: return queryRes` then the compiler will fail with:
```
Error: type mismatch: got <Future[Result[Option[seq[byte]], ref CatchableError]], Result[system.bool, ref CatchableError]> ... but expression 'queryRes' is of type: Result[system.bool, ref CatchableError]
```

Reply:
You are right - there are places where failure types are different. But in the places when thay match, I propose to return the same failure object rather than making a new one from the failure message

Reply:
I reviewed all usage of `return`, `failure`, `success` and made relevant changes.